beacon_node/beacon_chain/src/attestation_verification.rs

```diff
@@ -266,6 +266,16 @@ pub enum Error {
         attestation: Hash256,
         expected: Option<Hash256>,
     },
+    /// The attestation conflicts with finalization, no need to propagate.
+    ///
+    /// ## Peer scoring
+    ///
+    /// It's unclear if this attestation is valid, but it conflicts with finality and shouldn't be
+    /// propogated.
+    NotFinalizedDescendant {
+        attestation_block_root: Hash256,
+        attestation_slot: Slot,
+    },
```

Comment:
We already have a very similar error in the form of `HeadBlockFinalized`. I think it would be cleaner to use that